src/bp/ui-studio/src/web/views/OneFlow/sidePanel/form/SaySomethingForm.tsx

```diff
@@ -0,0 +1,244 @@
+}
+
+type StateProps = ReturnType<typeof mapStateToProps>
+type DispatchProps = typeof mapDispatchToProps
+type Props = DispatchProps & StateProps & OwnProps
+
+export interface FormState {
+  contentType: string
+  error: any
+}
+
+const shownCategories = ['builtin_text', 'builtin_image', 'builtin_carousel', 'builtin_card']
+const defaultFormState: FormState = {
+  contentType: 'builtin_text',
+  error: null
+}
+
+const SaySomethingForm: FC<Props> = props => {
+  const formReducer = (state: FormState, action): FormState => {
+    if (action.type === 'resetData') {
+      return {
+        ...state,
+        error: null,
+        contentType: 'builtin_text'
+      }
+    } else if (action.type === 'newData') {
+      const { contentType } = action.data
+
+      return {
+        error: null,
+        contentType: contentType || 'builtin_text'
+      }
+    } else if (action.type === 'updateContentType') {
+      const { value, initial } = action.data
+      const contentType = { contentType: value || 'builtin_text' }
+
+      if (!initial || !state.contentType) {
+        props.updateNode(contentType)
+      }
+
+      return {
+        ...state,
+        ...contentType
+      }
+    } else {
+      throw new Error(`That action type isn't supported.`)
+    }
+  }
+
+  const [formState, dispatchForm] = useReducer(formReducer, defaultFormState)
+  const [showOptions, setShowOptions] = useState(false)
+
+  useEffect(() => {
+    handleContentTypeChange(currentFlowNode?.contentType, true)
+    if (!props.categories?.length) {
+      props.fetchContentCategories()
+    }
+  }, [props.currentFlowNode.id])
+
+  const renameNode = text => {
+    if (text) {
+      const alreadyExists = props.currentFlow.nodes.find(x => x.name === text)
+
+      if (!alreadyExists) {
+        props.updateNode({ name: text })
+      }
+    }
+  }
+
+  const transformText = text => {
+    return text.replace(/[^a-z0-9-_\.]/gi, '_')
+  }
+
+  const onCopy = () => {
+    props.copyFlowNode()
+    setShowOptions(false)
+    toastInfo('Copied to buffer')
+  }
+
+  const handleContentTypeChange = (value, initial = false) => {
+    dispatchForm({ type: 'updateContentType', data: { value, initial } })
+  }
+
+  const { currentFlowNode, readOnly } = props
+  const { contentType } = formState
+  const categories = props.categories?.filter(cat => shownCategories.includes(cat.id))
+
+  const moreOptionsItems: MoreOptionsItems[] = [
+    {
+      icon: 'duplicate',
+      label: 'Copy',
+      action: onCopy.bind(this)
+    },
+    {
+      icon: 'trash',
+      label: 'Delete',
+      action: props?.onDeleteSelectedElements,
+      className: MoreOptionsStyles.delete
+    }
+  ]
+
+  const goThroughObjectAndLeaveOutIndex = (properties, indexToRemove) => {
+    const returnObject = {}
+
+    Object.keys(properties).forEach(key => {
+      if (key !== indexToRemove) {
+        returnObject[key] =
+          Object.prototype.toString.call(properties[key]) === '[object Object]'
```

Comment:
_.isObject ?

Reply:
_.isObject will return true even if it's an array... this is the best way I've found to make sure it was a proper object. here's a reference https://levelup.gitconnected.com/javascript-check-if-a-variable-is-an-object-and-nothing-else-not-an-array-a-set-etc-a3987ea08fd7

Reply:
I've used a small workaround with _.isObject && !_.isArray